libmamba/src/specs/version_spec.cpp

```diff
@@ -264,11 +264,14 @@ fmt::formatter<mamba::specs::VersionPredicate>::format(
             }
             if constexpr (std::is_same_v<Op, VersionPredicate::compatible_with>)
             {
+                // Make sure to print the version without loosing information.
+                auto version_level = pred.m_version.version().size();
+                auto format_level = std::max(op.level, version_level);
                 out = fmt::format_to(
                     out,
                     "{}{}",
                     VersionSpec::compatible_str,
-                    pred.m_version.str(op.level)
```

Comment:
Note to reviewer: this line cause the bug as the solver relies on the valid assumption that:

```c++
std::string match_spec_as_str = "numpy~=1.26"; // for instance.
match_spec_as_str == MatchSpec::parse(match_spec_as_str).value().str()
```

here:

https://github.com/mamba-org/mamba/blob/ad9b2d6b8840825411a3c23251cfbb978ab60e89/libmamba/src/solver/libsolv/helpers.cpp#L860

which was not verified for compatible versions.

I think we need to add tests cases (in another PR) to test that this is valid for many `MatchSpec`.